synapse/storage/schema/main/delta/72/04drop_column_application_services_state_last_txn.sql.sqlite

```diff
@@ -0,0 +1,42 @@
+/* Copyright 2022 The Matrix.org Foundation C.I.C
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+-- Drop unused column application_services_state.last_txn
+
+CREATE TABLE application_services_state2 (
+    as_id TEXT PRIMARY KEY,
```

Comment:
we should probably take the opportunity to make this `NOT NULL`. It's already `NOT NULL` on postgres as a side-effect of being a `PRIMARY KEY`, but sqlite doesn't do the same. (https://www.sqlite.org/lang_createtable.html#the_primary_key mentions this.)

Reply:
Fascinating thanks for sharing that link!